jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -1145,18 +1150,7 @@ public Throwable consumeAvailable()
         {
             Throwable result = HttpStream.consumeAvailable(this, getHttpConfiguration());
             if (result != null)
-            {
                 _generator.setPersistent(false);
-                // If HttpStream.consumeAvailable() returns an error, there may be unconsumed content left,
-                // so we must make sure the buffer is released and that the next chunk indicates the end of the stream.
-                if (_retainableByteBuffer != null)
-                {
-                    _retainableByteBuffer.release();
-                    _retainableByteBuffer = null;
-                }
-                if (_chunk == null)
-                    _chunk = Content.Chunk.from(result, true);
-            }
```

Comment:
I would restore this code for now, I don't think it can be safely removed.

As said in a previous comment, I'd like to see consistent handling of consumeAvailable() across protocols.

Reply:
Yes, I lost track of your original remark and got lost. This is now done.